android/compose/src/main/kotlin/com/boswelja/ephemeris/compose/InfiniteHorizontalPager.kt

```diff
@@ -87,43 +90,46 @@ public fun InfiniteHorizontalPager(
                             pagerHeight = pageHeight
                         }
                     }
+                    .padding(contentPadding)
             ) {
                 content(state.calculatePageFromInternal(index))
             }
         }
     }
 }
 
-public abstract class InfinitePagerState {
-
-    public abstract val pageCount: Int
-
-    public abstract var page: Int
-        internal set
-
-    internal abstract val internalStartPage: Int
+internal class InfinitePagerState internal constructor(
+    startPage: Int,
+    internal val lazyListState: LazyListState
+) {
 
-    internal abstract fun calculatePageFromInternal(internalPage: Int): Int
-}
+    val maxPages: Int = Int.MAX_VALUE
 
-internal class DefaultInfinitePagerState(
-    startPage: Int
-) : InfinitePagerState() {
+    val internalPageOffset: Int = maxPages / 2
 
-    override val internalStartPage: Int = Int.MAX_VALUE / 2
+    var page: Int by mutableStateOf(startPage)
 
-    override val pageCount: Int = Int.MAX_VALUE
+    fun calculatePageFromInternal(internalPage: Int): Int {
+        return internalPage - internalPageOffset
+    }
 
-    override var page: Int by mutableStateOf(startPage)
+    suspend fun scrollToPage(page: Int) {
+        lazyListState.scrollToItem(page + internalPageOffset)
+    }
 
-    override fun calculatePageFromInternal(internalPage: Int): Int {
-        return internalPage - internalStartPage
+    suspend fun animateScrollToPage(page: Int) {
+        lazyListState.animateScrollToItem(page + internalPageOffset)
     }
 }
 
 @Composable
-public fun rememberInfinitePagerState(startPage: Int = 0): InfinitePagerState {
+internal fun rememberInfinitePagerState(
+    startPage: Int = 0
```

Comment:
this is internal now, so there's no way for a user to start on a page that isn't 0?

Reply:
In the context of Ephemeris calendar Composable, the start page is dictated by the page source rather than the page number. If users wanted to implement InfoniteHorizontalPager then that's another story :P

Reply:
So this is just a page 0 to enable the InfinitePager to work, and not actually a calendar page?

Reply:
Yep, pretty much

Reply:
Cool, will resolve with no fix planned then